packages/terra-application-utility/src/utility/_UtilityMenu.jsx

```diff
@@ -0,0 +1,319 @@
+        isSelectable: item.isSelectable,
+        childKeys: item.childKeys,
+      },
+    );
+  });
+  return map;
+};
+
+const hasChevron = item => item.childKeys && item.childKeys.length > 0;
+
+const contextTypes = {
+  /* eslint-disable consistent-return */
+  intl: (context) => {
+    if (context.intl === undefined) {
+      return new Error('Please add locale prop to Base component to load translations');
+    }
+  },
+};
+
+class UtilityMenu extends React.Component {
+  constructor(props) {
+    super(props);
+    this.getItem = this.getItem.bind(this);
+    this.buildItem = this.buildItem.bind(this);
+    this.buildListContent = this.buildListContent.bind(this);
+    this.buildFooterContent = this.buildFooterContent.bind(this);
+    this.childrenHasCheckmark = this.childrenHasCheckmark.bind(this);
+    this.childrenHasChevron = this.childrenHasChevron.bind(this);
+    this.handleOnChange = this.handleOnChange.bind(this);
+    this.handleRequestBack = this.handleRequestBack.bind(this);
+    this.pop = this.pop.bind(this);
+    this.push = this.push.bind(this);
+    this.state = {
+      map: processMenuItems(props.menuItems),
+      currentKey: props.selectedKey,
+      previousKeyStack: [],
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.menuItems === this.props.menuItems) {
+      return;
+    }
+
+    this.setState({
+      map: processMenuItems(nextProps.menuItems),
+    });
+  }
+
+  getItem(key) {
+    return this.state.map.get(key);
+  }
+
+  buildItem(key, leftInset, rightInset) {
+    const item = this.getItem(key);
+    const chevron = hasChevron(item);
+    return (
+      <MenuItem
+        key={key}
+        itemKey={key}
+        id={item.id}
+        title={item.title}
+        content={item.content}
+        contentLocation={item.contentLocation}
+        isSelected={item.isSelected}
+        isSelectable={item.isSelectable}
+        hasChevron={chevron}
+        leftInset={leftInset}
+        rightInset={rightInset}
+        onChange={this.handleOnChange}
+        variant={this.props.variant}
+      />
+    );
+  }
+
+  buildListContent(currentItem) {
+    if (currentItem && currentItem.childKeys && currentItem.childKeys.length) {
+      const leftInset = this.childrenHasCheckmark(currentItem);
+      const rightInset = this.childrenHasChevron(currentItem);
+      return (
+        <ul className={cx('body')}>
+          {currentItem.childKeys.map((key) => {
+            if (this.getItem(key).contentLocation !== Utils.LOCATIONS.FOOTER) {
+              return this.buildItem(key, leftInset, rightInset);
+            }
+            return null;
+          })}
+        </ul>
+      );
+    }
+    return null;
+  }
+
+  buildFooterContent(currentItem) {
+    if (currentItem && currentItem.childKeys && currentItem.childKeys.length) {
+      return currentItem.childKeys.map((key) => {
+        let item = null;
+        if (this.getItem(key).contentLocation === Utils.LOCATIONS.FOOTER) {
```

Comment:
You don't have to but you could avoid the variable by returning this.buildItem(key) on line 153 and null on 155